### PR TITLE
Module3, task

### PR DESCRIPTION
**Pull request by Nastya-Korsak**
Module 3, fix tests

### CODE REVIEW

File: Module_3/DebuggingFundamentals/Task1/Utilities.cs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        /// Searches for the index of a product in an <paramref name="products"/> 
+        /// based on a <paramref name="predicate"/>.
+        /// </summary>
+        /// <param name="products">Products used for searching.</param>
+        /// <param name="predicate">Product predicate.</param>
+        /// <returns>If match found then returns index of product in <paramref name="products"/>
+        /// otherwise -1.</returns>
+        public static int IndexOf(Product[] products, Predicate<Product> predicate)
+        {
+            if (products == null || predicate == null)
```
Comment (alexander-trishin) [inline Module_3/DebuggingFundamentals/Task1/Utilities.cs:43]: Please check for null separately and don't forget to specify the argument name in `ArgumentNullException`.

**Review by alexander-trishin — CHANGES_REQUESTED** [review]
Hi Nastya 😃

In general, great job! Please fix the code, according to my comments and we are good to go.

File: Module_3/DebuggingFundamentals/Task1/Product.cs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        public string Name { get; set; }
+
+        public double Price { get; set; }
+
+        public override bool Equals(object obj)
+        {
+            return obj is Product product && Equals(product);
+        }
+
+        public bool Equals(Product other)
```
[the remark this answers is not in the document]
Comment (Nastya-Korsak) [reply Module_3/DebuggingFundamentals/Task1/Product.cs:22]: fixed

File: Module_3/DebuggingFundamentals/Task1/Task1.csproj [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <CodeAnalysisRuleset>..\..\..\root.ruleset</CodeAnalysisRuleset>
+  </PropertyGroup>
```
[the remark this answers is not in the document]
Comment (Nastya-Korsak) [reply Module_3/DebuggingFundamentals/Task1/Task1.csproj:6]: added

File: Module_3/DebuggingFundamentals/Task1/Utilities.cs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    {
+        /// <summary>
+        /// Sorts an array in ascending order using bubble sort.
+        /// </summary>
+        /// <param name="numbers">Numbers to sort.</param>
+        public static void Sort(int[] numbers)
+        {
+            if (numbers == null)
+            {
+                throw new ArgumentNullException();
```
[the remark this answers is not in the document]
Comment (Nastya-Korsak) [reply Module_3/DebuggingFundamentals/Task1/Utilities.cs]: fixed

File: Module_3/DebuggingFundamentals/Task1/Utilities.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+            }
+
+            int temp;
```
[the remark this answers is not in the document]
Comment (Nastya-Korsak) [reply Module_3/DebuggingFundamentals/Task1/Utilities.cs]: moved

File: Module_3/DebuggingFundamentals/Task1/Utilities.cs (hunk 1 above)
Comment (Nastya-Korsak) [reply Module_3/DebuggingFundamentals/Task1/Utilities.cs]: done

**Review by alexander-trishin — APPROVED** [review]

**Pull request merged**